cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/controllers/CatsSqlAdminController.kt

```diff
@@ -102,7 +93,39 @@ class CatsSqlAdminController(
     return CleanTablesResult(tableCount = tablesDropped.size, tables = tablesDropped)
   }
 
-  private fun validateParams(currentNamespace: String?, targetNamespace: String) {
+  @PutMapping(path = ["drop_version/{version}"])
+  fun dropTablesByVersion(@PathVariable("version") dropVersion: String): CleanTablesResult {
+    validatePermissions()
+
+    val conn = getConnection()
+    val sql = "show tables like 'cats_v$dropVersion%"
```

Comment:
Two things here:
(1) Even on an admin endpoint I'd avoid directly interpolating user input into a SQL query like this.  Does JOOQ support binding a variable into this query?
(2) What happens when we're on version 10 and someone tries to drop version 1?

Reply:
By binding a variable, I mean that in SQL supports distinguishing what is part of the query syntax from what is a value that is passed to that syntax (and this support is included in most libraries for sending SQL queries, including JOOQ).

So for example, if I wanted to fetch all documents with `owner = 'eric'`, one way to send the query would be:
```
val user = "eric"
jooq.fetch("select * from document where owner = '$user'")
```

Alternatively, you could pull the variable into a bind, and send:
```
jooq.fetch("select * from document where owner = ?", $user)
```

For security reasons, the second of these is much better, because SQL understands that the value is not part of the query and does not consider it when parsing the query.  So even if you passed unsanitized user input as a bind, it could not perform SQL injection because the query itself is fixed, it's only the bind variable that can change.

In the first case, if `user` ends up being set to `eric' or 1=1` then the query will return all rows from the database.  In the second case, it would return all rows whose owner is equal to the string `eric' or 1=1` which is what the code intends (and foils the malicious attempt to dump out the whole database).

The other reason binds are better is that they often lead to better performance, because the database needs to parse fewer queries.  Without binds, the database needs to separately parse the text of `select * from document where owner = 'eric'` and `select * from document where owner = 'matt'` (and presumably the many other versions of this query if it exists in my application logic).  With binds, it only needs to parse the single query `select * from document where owner = ?` and can cache the execution plan for the query (basically compiled database code) and then re-use that compiled plan every time it runs with a different bind variable.

Neither of these are the most important thing in this case, as it's a lightly-used page that only admins can access, but those are the general reasons why using bind variables is good practice.  (And the reasons why I get nervous when I see queries that don't use binds.)

I think here something like `jooq.execute("show tables like", "cats_v${drop_version}_%")` would be the way to bind it.

Reply:
Thank you so much for the informative response. I'll update to make use of the [fetch with bindings](https://www.jooq.org/javadoc/latest/org.jooq/org/jooq/DSLContext.html#fetch(java.lang.String,java.lang.Object...)) and while I'm at it update the other 2 exposed admin endpoints to do the same.